src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -153,15 +153,15 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 // Dictionary does not guarantee order will be preserved.
                 var groupedDisplayNamesAndValues = new List<KeyValuePair<EnumGroupAndName, string>>();
                 var namesAndValues = new Dictionary<string, string>();
+                IStringLocalizer enumLocalizer = _stringLocalizerFactory?.Create(underlyingType);
                 foreach (var name in Enum.GetNames(underlyingType))
                 {
                     var field = underlyingType.GetField(name);
-                    var displayName = GetDisplayName(field);
                     var groupName = GetDisplayGroup(field);
                     var value = ((Enum)field.GetValue(obj: null)).ToString("d");
 
                     groupedDisplayNamesAndValues.Add(new KeyValuePair<EnumGroupAndName, string>(
-                        new EnumGroupAndName(groupName, displayName),
+                        new EnumGroupAndName(groupName, () => GetDisplayName(enumLocalizer, field)),
```

Comment:
Check whether this capturing lambda chews through more memory than instantiating the `EnumGroupAndName` with the `IStringLocalizer` and `FieldInfo`. Don't need `GetDisplayName()` in this class if that works better.